### DISCUSSION TITLE
[useSWRInfinite] Pass last item of `initialData` to  first call of `getKey`

### DISCUSSION THREAD

**Discussion by akellbl4** [vercel/swr#599] [General]
I'm using NextJS and want to create 'load more' pagination with `useSWRInfinite`.
But it is strange behavior for me when I provide `initialData` and than `getKey` calls with `page: 0` and `prevPageData: null`

I expect if I pass initial data in count of 1 item it will pass `page` as 1 and my initial data as `prevPageData` because I already have first loaded page.

For example:
```typescript
type 
const getKey = (page: number, previousPageData: { items: HomeItem[]; next: null | number }) => {
	if (!previousPageData) return `/api/homes?page=1`
	if (previousPageData.next === null) return null
	return `/api/homes?page=${previousPageData.next}`
}

const MyComponent: NextPage<{ items: HomeItem[]; next: null | number }> = ({ items, next }) => {
	const { data, size, setSize } = useSWRInfinite<{ items: HomeItem[]; next: null | number }>(getKey, fetcher, {
		initialData: [{ items, next }],
	})
	
	...
}
```

**Comment by Xetera** [comment]
Were you able to solve this issue? Seems like it's still not fixed
